modules/contact-form/grunion-contact-form.php

```diff
@@ -3513,3 +3517,63 @@ function grunion_delete_old_spam() {
 		wp_schedule_single_event( time() + 700, 'grunion_scheduled_delete' );
 	}
 }
+
+/**
+ * Send an event to Tracks on form submission.
+ *
+ * @param int   $post_id - the post_id for the CPT that is created.
+ * @param array $all_values - fields from the default contact form.
+ * @param array $extra_values - extra fields added to from the contact form.
+ *
+ * @return null|void
+ */
+function jetpack_tracks_record_grunion_pre_message_sent( $post_id, $all_values, $extra_values ) {
+	// Do not do anything if the submission is not from a block.
+	if (
+		! isset( $extra_values['is_block'] )
+		|| ! $extra_values['is_block']
+	) {
+		return;
+	}
+
+	/*
+	 * Event details.
+	 */
+	$event_user  = wp_get_current_user();
+	$event_name  = 'contact_form_block_message_sent';
+	$event_props = array(
+		'entry_permalink' => esc_url( $all_values['entry_permalink'] ),
+		'feedback_id'     => esc_attr( $all_values['feedback_id'] ),
+	);
+
+	/*
+	 * Record event.
+	 * We use different libs on wpcom and Jetpack.
+	 */
+	if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+		$event_name             = 'wpcom_' . $event_name;
+		$event_props['blog_id'] = get_current_blog_id();
+		// If the form was sent by a logged out visitor, record event with blog owner.
+		if ( empty( $event_user->ID ) ) {
+			$event_user_id = wpcom_get_blog_owner( $event_props['blog_id'] );
+			$event_user    = get_userdata( $event_user_id );
+		}
+
+		require_lib( 'tracks/client' );
+		tracks_record_event( $event_user, $event_name, $event_props );
+	} else {
+		$event_name             = 'jetpack_' . $event_name;
+		$event_props['blog_id'] = Jetpack_Options::get_option( 'id', 0 );
+		// If the form was sent by a logged out visitor, record event with Jetpack master user.
+		if ( empty( $event_user->ID ) ) {
+			$master_user_id = Jetpack_Options::get_option( 'master_user' );
+			if ( ! empty( $master_user_id ) ) {
+				$event_user = get_userdata( $master_user_id );
+			}
+		}
+
+		$tracking = new Automattic\Jetpack\Tracking();
+		$tracking->tracks_record_event( $event_user, $event_name, $event_props );
```

Comment:
You may be able to instead use `record_user_event()` which will handle some of the logic you've written such as the blog_id and the `jetpack_` prefix to the event name.

Reply:
Done in 012227f93e73f20984eec27d9f1096e929116375. I also updated the wpcom diff to match this.